lib/improver/tests/nowcasting/optical_flow/test_AdvectField.py

```diff
@@ -396,6 +405,26 @@ def test_lead_time(self):
         self.assertEqual(len(lead_time), 1)
         self.assertEqual(lead_time[0], self.timestep.total_seconds())
 
+    def test_units_and_datatypes_for_time_coordinates(self):
+        """Test that the output cube contains the desired units and datatypes
+        for the time, forecast_reference_time and forecast_period coordinate.
+        """
```

Comment:
We also need an additional test that data coming in with time coordinates in hours will have the correct output coordinate values and datatypes.  You could do this using "self.cube" via `self.cube.coord("time").convert_units("hours since 1970-01-01 00:00")` etc.

Reply:
I've added a unit test, as suggested, where the original time coordinate is in the units of "hours since 1970".